## 타이타닉 생존자에 관한 보고서
- 1. 타이타닉에 생존한 사람 조사
- 2. 여자, 남자, 아이 를 비교해서 누가 더 많이 살아 남았을까?
- 3. 많이 생존한 쪽은 왜 많이 생존할 수 있었을까?
- 4. 적게 생존한 쪽은 왜 적게 생존 했을까?

In [183]:
import pandas as pd
import numpy as np

In [184]:
# 데이터 준비
file_name = 'titanic.csv.csv'
titanic=pd.read_csv(file_name)
print(titanic)

     생존자  티켓등급      성별    나이  형제,배우자  부모,자녀       요금 탑승지역      등급    who  \
0      0     3    male  22.0       1      0   7.2500    S   Third    man   
1      1     1  female  38.0       1      0  71.2833    C   First  woman   
2      1     3  female  26.0       0      0   7.9250    S   Third  woman   
3      1     1  female  35.0       1      0  53.1000    S   First  woman   
4      0     3    male  35.0       0      0   8.0500    S   Third    man   
..   ...   ...     ...   ...     ...    ...      ...  ...     ...    ...   
886    0     2    male  27.0       0      0  13.0000    S  Second    man   
887    1     1  female  19.0       0      0  30.0000    S   First  woman   
888    0     3  female   NaN       1      2  23.4500    S   Third  woman   
889    1     1    male  26.0       0      0  30.0000    C   First    man   
890    0     3    male  32.0       0      0   7.7500    Q   Third    man   

      성인남자   위치       탑승지역.1   생존   결혼안함  
0     True  NaN  Southampton   no  False  
1

In [185]:
# 데이터 전처리
# 타이타닉 생존자열 추출
sangzon=pd.Series(titanic['생존자'])

In [186]:
# 891명 중에 생존자 사망자 값
titanic['생존자'].value_counts(0)
titanic['who'].value_counts(0)
# 생존자 : 342 생존률 : 38.38%
# 사망자 : 549 사망률 : 61.62%

who
man      537
woman    271
child     83
Name: count, dtype: int64

In [187]:
# 타이타닉 데이터 프레임을 생존자만 나타나게 만들기
sangzon=titanic['생존자']>0
sanja=titanic[sangzon]
sanja

생존자  티켓등급      성별    나이  형제,배우자  부모,자녀       요금 탑승지역      등급    who  \
1      1     1  female  38.0       1      0  71.2833    C   First  woman   
2      1     3  female  26.0       0      0   7.9250    S   Third  woman   
3      1     1  female  35.0       1      0  53.1000    S   First  woman   
8      1     3  female  27.0       0      2  11.1333    S   Third  woman   
9      1     2  female  14.0       1      0  30.0708    C  Second  child   
..   ...   ...     ...   ...     ...    ...      ...  ...     ...    ...   
875    1     3  female  15.0       0      0   7.2250    C   Third  child   
879    1     1  female  56.0       0      1  83.1583    C   First  woman   
880    1     2  female  25.0       0      1  26.0000    S  Second  woman   
887    1     1  female  19.0       0      0  30.0000    S   First  woman   
889    1     1    male  26.0       0      0  30.0000    C   First    man   

      성인남자   위치       탑승지역.1   생존   결혼안함  
1    False    C    Cherbourg  yes  False  
2    False  NaN  Southampton  yes   True  
3    False    C  Southampton  yes  False  
8    False  NaN  Southampton  yes  False  
9    False  NaN    Cherbourg  yes  False  
..     ...  ...          ...  ...    ...  
875  False  NaN    Cherbourg  yes   True  
879  False    C    Cherbourg  yes  False  
880  False  NaN  Southampton  yes  False  
887  False    B  Southampton  yes   True  
889   True    C    Cherbourg  yes   True  

[342 rows x 15 columns]

In [188]:
sanja=sanja.drop(['티켓등급', '나이', '형제,배우자', '부모,자녀', '요금','탑승지역','등급','성인남자','위치','탑승지역.1','생존','결혼안함'],axis=1)
sanja

생존자      성별    who
1      1  female  woman
2      1  female  woman
3      1  female  woman
8      1  female  woman
9      1  female  child
..   ...     ...    ...
875    1  female  child
879    1  female  woman
880    1  female  woman
887    1  female  woman
889    1    male    man

[342 rows x 3 columns]

In [189]:
sangzonza=list(sanja['생존자'].value_counts(0))
sungbae=list(sanja['성별'].value_counts())
# 여성 : 233명
# 남성 : 109명

In [190]:
who=list(sanja['who'].value_counts())
# 여성 : 205명
# 남성 : 88명
# 어린이 : 49명

In [191]:
# value_count 값을 Series로 정리하기
sangzonza2=pd.Series(sangzonza)
sungbae2=pd.Series(sungbae)
who2=pd.Series(who)
print(sangzonza2)
print(sungbae2)
print(who2)

0    342
dtype: int64
0    233
1    109
dtype: int64
0    205
1     88
2     49
dtype: int64


In [192]:
# 정리한 시리즈를 DataFrame으로 만들기
dataframe=pd.DataFrame({
    
    '남녀생존' : sungbae2,
    '성인과아이' : who2
})

In [193]:
dataframe

남녀생존  성인과아이
0  233.0    205
1  109.0     88
2    NaN     49

In [194]:
# 행인덱스 변경 0,1,2 --> 남자, 여자, 아이로 변경
dataframe.index=['여자','남자','아이']
dataframe

남녀생존  성인과아이
여자  233.0    205
남자  109.0     88
아이    NaN     49

In [195]:
dataframe.fillna(0, inplace=True)

In [196]:
# 총 생존자 행 추가
dataframe.loc['총생존자']=[342,'전체인원']

In [197]:
# 전체 인원 넣기
# who
# man      537
# woman    271
# child     83

In [198]:
dataframe['전체인원']=[271,537,83,891]
dataframe

남녀생존 성인과아이  전체인원
여자    233.0   205   271
남자    109.0    88   537
아이      0.0    49    83
총생존자  342.0  전체인원   891

In [199]:
dataframe['인원대비 생존률']=['75.65%','20.30%','59.04%','비고']

In [200]:
dataframe

남녀생존 성인과아이  전체인원 인원대비 생존률
여자    233.0   205   271   75.65%
남자    109.0    88   537   20.30%
아이      0.0    49    83   59.04%
총생존자  342.0  전체인원   891       비고

위 결과를 보시면 인원대비 여성의 생존률이 앞도적으로 높습니다.
여성은 75%의 생존률
남성은 20%의 생존률
아이는 59%의 생존률이였습니다.
어떻게 왜 여성이 많이 생존 할 수 있었을까요?

당시 사회적 규범과 해양 전통에는 남성들의 '신사, 기사도' 문화로 인해 여성들에게 양보하고 희생하는 경향이 있었습니다.
그래서 재난 상황에서 여성과 어린이를 먼저 구조하려 했던 경향이 많았습니다.
그리고 타이타닉 호가 침몰할 때, 승무원들에게 여성과 어린이를 먼저 구명보트에 태우라는 방송이 울렸고, 사람들은 명령에 충실히 임했습니다.
남성들의 생존률은 20% 대로 충분하지 않은 구명보트를 여성들에게 먼저 양보 하는 경우가 많았습니다.
그래서 남성들은 여성과 어린이들을 먼저 태운후 남은 자리를 차지했습니다.
아이들의 생존률도 60%에 가까웠던 이유가 남성들이 그들을 제일 먼저 구하고 자신을 희생했기에 여성 다음으로 높은 생존률을 보였습니다.
이들의 높은 기사도 정신과 희생정신으로 인해 "여성 및 어린이 우선" 이라 생각하고 행동한 남성들이 많았습니다.
위 결과는 당시 사회적 규범과 구조적요인, 사고 당시 구체적인 상황들이 복합적으로 작용한 결과입니다.
안타깝게도 남성들의 생존률이 낮았던 이유는 구명보트의 부족과 그들의 희생정신이 크게 작용하여 생존자가 20%에 머무른 모습이였습니다.
사고로 목숨을 잃은 사람들과 자신을 희생하여 소중한 생명을 살렸던 사람들에게 존경과 명복을 빕니다.
